develocity-gradle.yml

```diff
@@ -52,41 +50,49 @@ spec:
 
         def atLeastGradle3 = version >= GradleVersion.version("3.0")
         def atLeastGradle6 = version >= GradleVersion.version("6.0")
+        def captureMarker = new AtomicBoolean(false)
 
         if (atLeastGradle6) {
-          def useBuildService = version >= GradleVersion.version("6.6")
           settingsEvaluated { settings ->
+            settings.pluginManager.withPlugin("com.gradle.develocity") {
+              captureUsingBuildScanPublished(settings.extensions["develocity"].buildScan, settings.rootProject, captureMarker)
+            }
             settings.pluginManager.withPlugin("com.gradle.enterprise") {
-              captureUsingBuildScanPublished(settings.extensions["gradleEnterprise"].buildScan, settings.rootProject)
+              captureUsingBuildScanPublished(settings.extensions["gradleEnterprise"].buildScan, settings.rootProject, captureMarker)
             }
           }
         } else if (atLeastGradle3) {
           projectsEvaluated { gradle ->
+            gradle.rootProject.pluginManager.withPlugin("com.gradle.develocity") {
+              captureUsingBuildScanPublished(gradle.rootProject.extensions["develocity"].buildScan, gradle.rootProject, captureMarker)
+            }
             gradle.rootProject.pluginManager.withPlugin("com.gradle.build-scan") {
-              captureUsingBuildScanPublished(gradle.rootProject.extensions["buildScan"], gradle.rootProject)
+              captureUsingBuildScanPublished(gradle.rootProject.extensions["buildScan"], gradle.rootProject, captureMarker)
             }
           }
         }
       }
 
-      def captureUsingBuildScanPublished(buildScanExtension, rootProject) {
-        buildScanExtension.with {
-          buildScanPublished { buildScan ->
-            if (System.getenv("BUILD_SCAN_REPORT_PATH")) {
-              def reportFile = new File(System.getenv("BUILD_SCAN_REPORT_PATH"))
-              def report
-              // This might have been created by a previous Gradle invocation in the same GitLab job
-              // Note that we do not handle parallel Gradle scripts invocation, which should be a very edge case in context of a GitLab job
-              if (reportFile.exists()) {
-                report = new groovy.json.JsonSlurper().parseText(reportFile.text) as Report
-              } else {
-                report = new Report()
+      def captureUsingBuildScanPublished(buildScanExtension, rootProject, AtomicBoolean captureMarker) {
+        if (captureMarker.compareAndSet(false, true)) {
```

Comment:
🤔 What's the purpose of this `captureMarker` ? I don't think it would protect from 2 parallel Gradle builds as this var won't be shared between Gradle executions, right ?

Reply:
No, it wouldn't. The rationale is that the current version of the Develocity plugin will transitively apply the GE plugin. It means that this callback will be registered twice: once as `Develocity#buildScanPublished` and once as `GE#buildScanPublished`. We don't really have a way to deduplicate them on the plugin side. So, this marker makes sure we execute the action just once in this build.